cpp/src/parquet/arrow/test_util.h

```diff
@@ -463,12 +463,16 @@ Status MakeEmptyListsArray(int64_t size, std::shared_ptr<Array>* out_array) {
   return Status::OK();
 }
 
+// Please make sure that the return value should have an object assigned explicitly
+// if you want to use the pointer contained.
```

Comment:
I don't understand what this comment means, nor why you are adding it in multiple places. Would you like to explain?

Reply:
Hi pitrou. I think I misunderstood the meaning of [emkornfield](https://github.com/emkornfield). According to https://github.com/apache/arrow/issues/14748#issuecomment-1333336429, I thought that when a function returns a shared_ptr containing a new object (the pointer contained is allocated in the function), then it need to add the comment "sth. must not outlive some object". This function is static, which belongs to no object. So, I put my word to replace that comment. And you can refer to #14748, I open that issue because I directly used the bare pointer without explicitly assignment to an object, and it causes the access violation.

Reply:
Does that make sense? And is it necessary to make this PR? I think it was my problem that triggers the access violation, but not the docs.

Reply:
sorry for chiming, in there were at least a few places that seemed to be incorrect on specifying the the independence of the shared_ptr, I think my intent was really only places in column_reader.h.

Reply:
Oh, really sorry, it looks like I truly misunderstood your meaning. I thought that you wanted me to verify all the ownership of functions that return a shared_ptr. I'll have a careful check and then make a change on this PR.